src/main/java/com/google/cloud/healthcare/etl/runner/dicomtofhir/DicomToFhirStreamingRunner.java

```diff
@@ -0,0 +1,241 @@
+package com.google.cloud.healthcare.etl.runner.dicomtofhir;
+
+import com.google.cloud.healthcare.etl.model.ErrorEntry;
+import com.google.cloud.healthcare.etl.model.converter.ErrorEntryConverter;
+import com.google.cloud.healthcare.etl.model.mapping.HclsApiDicomMappableMessage;
+import com.google.cloud.healthcare.etl.model.mapping.HclsApiDicomMappableMessageCoder;
+import com.google.cloud.healthcare.etl.pipeline.MappingFn;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.io.gcp.healthcare.DicomIO;
+import org.apache.beam.sdk.io.gcp.healthcare.FhirIO;
+import org.apache.beam.sdk.io.gcp.healthcare.HealthcareIOError;
+import org.apache.beam.sdk.io.gcp.healthcare.HealthcareIOErrorToTableRow;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubIO;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.Validation.Required;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.windowing.AfterProcessingTime;
+import org.apache.beam.sdk.transforms.windowing.FixedWindows;
+import org.apache.beam.sdk.transforms.windowing.Repeatedly;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.*;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * The entry point of the pipeline. It will be triggered upon receiving a PubSub message from the given subscription. From
+ * the message, it will extract the webpath  of a DICOM instance and collect it's metadata from DICOM IO. With the
+ * response, it will map the study metadata to a FHIR ImagingStudyResource and upload it to the given FHIR store.
+ *
+ * The errors for each component are handled separately, e.g. you can specify file paths for each
+ * of the stage (read - DICOM IO, mapping, write - FHIR IO). Right now the shard is set to 1, if
+ * you are seeing issues with regard to writing to GCS, feel free to bump it up to a reasonable
+ * value.
+ *
+ */
+public class DicomToFhirStreamingRunner {
+    private static final Logger LOG = LoggerFactory.getLogger(DicomToFhirStreamingRunner.class);
+    private static Duration ERROR_LOG_WINDOW_SIZE = Duration.standardSeconds(5);
+
+    /**
+    * A DoFn to collect the webpath of the DICOM instance from the PubSub Message.
+     */
+    public static class ExtractWebpathFromPubsub extends DoFn<PubsubMessage, String> {
```

Comment:
Does this class need to be public?

Reply:
It does because there is a unit test that requires it to be public

Reply:
Can it be package private rather than public (e.g. no modifier)?

Reply:
Yep, that helps. I didn't put the test files in the right package so it was throwing an error. TIL how src/main and src/test packaging works :)